pkg/cri/opts/spec_windows.go

```diff
@@ -0,0 +1,121 @@
+//go:build windows
+
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package opts
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"github.com/containerd/containerd/containers"
+	"github.com/containerd/containerd/oci"
+	imagespec "github.com/opencontainers/image-spec/specs-go/v1"
+	runtimespec "github.com/opencontainers/runtime-spec/specs-go"
+	"golang.org/x/sys/windows"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func escapeAndCombineArgsWindows(args []string) string {
```

Comment:
If this is the only function which has Windows specific logic, seems better to only have this in the windows only part of the build. The rest of the logic may be intended for windows, but with sbserver the logic does not need to be only compiled on Windows.

Reply:
escapeAndCombineArgsWindows() is called from WithProcessCommandLineOrArgs() which is the function below this, making it windows only. And the other function in this file getArgs() is a helper function called only from WithProcessCommandLineOrArgs() so it made more sense to have them all as part of the spec_windows.go file.

Reply:
@dmcgowan ping on this :)

Reply:
renamed WithProcessCommandLineOrArgs() function in this file to windows only - WithProcessCommandLineOrArgsForWindows(). the getArgs() funciton is the only other one in this file and it is a helper function called from WithProcessCommandLineOrArgsForWindows() only. Therefore leaving all these functions to this file as that seems to make more sense.